local/blockchain.go

```diff
@@ -702,49 +691,144 @@ func (ln *localNetwork) reloadVMPlugins(
 	for i, chainSpec := range chainSpecs {
 		vmName := chainSpec.VmName
 		vmID, err := utils.VMID(vmName)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
-		vmGenesisBytes := chainSpec.Genesis
+		genesisBytes := chainSpec.Genesis
 
 		log.Info("creating blockchain tx",
 			zap.String("vm-name", vmName),
 			zap.String("vm-ID", vmID.String()),
-			zap.Int("bytes length of genesis", len(vmGenesisBytes)),
+			zap.Int("bytes length of genesis", len(genesisBytes)),
 		)
 		cctx, cancel := createDefaultCtx(ctx)
+		defer cancel()
 		subnetID, err := ids.FromString(*chainSpec.SubnetId)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
-		blockchainID, err := baseWallet.P().IssueCreateChainTx(
+		utx, err := baseWallet.P().Builder().NewCreateChainTx(
 			subnetID,
-			vmGenesisBytes,
+			genesisBytes,
 			vmID,
 			nil,
 			vmName,
+		)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failure generating create blockchain tx: %w", err)
+		}
+		tx, err := baseWallet.P().Signer().SignUnsigned(cctx, utx)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failure signing create blockchain tx: %w", err)
+		}
+
+		blockchainIDs[i] = tx.ID()
+		blockchainTxs[i] = tx
+	}
+
+	return blockchainIDs, blockchainTxs, nil
+}
+
+func (ln *localNetwork) createBlockchainConfigFiles(
+	chainSpecs []network.BlockchainSpec,
+	blockchainIDs []ids.ID,
+	log logging.Logger,
+) (bool, error) {
+	println()
+	created := false
+	log.Info(logging.Green.Wrap("creating config files for each custom chain"))
+	for i, chainSpec := range chainSpecs {
+		chainAlias := blockchainIDs[i].String()
+
+		// create config and network upgrade files
+		if chainSpec.ChainConfig != nil {
+			created = true
+			for nodeName := range ln.nodes {
+				nodeRootDir := getNodeDir(ln.rootDir, nodeName)
+				chainConfigDir := filepath.Join(nodeRootDir, chainConfigSubDir)
+				chainConfigPath := filepath.Join(chainConfigDir, chainAlias, configFileName)
+				if err := createFileAndWrite(chainConfigPath, chainSpec.ChainConfig); err != nil {
+					return false, fmt.Errorf("couldn't write file at %q: %w", chainConfigPath, err)
+				}
+			}
+		}
+		if chainSpec.NetworkUpgrade != nil {
+			created = true
+			for nodeName := range ln.nodes {
+				nodeRootDir := getNodeDir(ln.rootDir, nodeName)
+				chainConfigDir := filepath.Join(nodeRootDir, chainConfigSubDir)
+				chainUpgradePath := filepath.Join(chainConfigDir, chainAlias, upgradeConfigFileName)
+				if err := createFileAndWrite(chainUpgradePath, chainSpec.NetworkUpgrade); err != nil {
+					return false, fmt.Errorf("couldn't write file at %q: %w", chainUpgradePath, err)
+				}
+			}
+		}
+		// update config info for snapshopt/restart purposes
+		// put into defaults and reset node specifics
+		if chainSpec.ChainConfig != nil {
+			ln.chainConfigFiles[chainAlias] = string(chainSpec.ChainConfig)
+			for nodeName := range ln.nodes {
+				delete(ln.nodes[nodeName].config.ChainConfigFiles, chainAlias)
```

Comment:
Why do we need this again?

Reply:
ln.chainConfigFiles[chainAlias] is setting the global network conf, which will be applied to current and added nodes.
as the user requested for the network to use that conf for that blockchain, remove any previous node-only conf related to it, so it will automatically peek the global one.